### PR TITLE
Allow plugins to report info such as version, report that in Status

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
Adds a virtual function to the `BedrockPlugin` class to allow plugins to report their version or other info about themselves.

Updates the `Status` plugin to report the compiled-in plugins, with their versions, if available. This results in output like (note: whitespace added for display, not in actual command output):

```
{
  "commitCount": 49459,
  "escalatedCommandList": [],
  "hash": "DCD509830940FE37E3152C72F1359ACFF7CE59E2",
  "isMaster": true,
  "peerList": [],
  "plugins": [
    {
      "name": "Cache"
    },
    {
      "name": "DB"
    },
    {
      "name": "Jobs"
    },
    {
      "name": "MySQL"
    },
    {
      "name": "Status"
    },
    {
      "name": "Auth",
      "version": "3df0282b8b1b20c6d37861ca56c45f8d4e810eec"
    }
  ],
  "priority": 200,
  "processedCommandList": [],
  "queuedCommandList": [],
  "state": "MASTERING",
  "version": "7cde42bf2afb103c164ad883e033027f79c97fbc"
}
```
The above was created with an experimental change to the Auth plugin, which will get it's own PR.

Tests:
No tests.

### CODE REVIEW

File: BedrockPlugin.cpp [hunk 1]
```diff
@@ -46,6 +46,7 @@ BedrockPlugin* BedrockPlugin::getPlugin(const string& name) {
 void BedrockPlugin::enable(bool enabled) { _enabled = enabled; }
 bool BedrockPlugin::enabled() { return _enabled; }
 string BedrockPlugin::getName() { SERROR("No name defined by this plugin, aborting."); }
+const STable& BedrockPlugin::getInfo() { return pluginInfo; }
```
Comment (cead22) [inline BedrockPlugin.cpp]: Nab just something to keep in mind, but doesn't this mean that Plugins will have to be married to STable (and thus libstuff) in order to report Status?

File: BedrockPlugin.h [hunk 2]
```diff
@@ -83,6 +86,8 @@ class BedrockPlugin {
     // Attributes
     bool _enabled;
 
+    STable pluginInfo;
```
Comment (quinthar) [inline BedrockPlugin.h]: Can you do `_pluginInfo` and remove the extra blank line?

File: BedrockPlugin.h [hunk 3]
```diff
@@ -28,6 +28,9 @@ class BedrockPlugin {
     // Returns a short, descriptive name of this plugin
     virtual string getName();
 
+    // Returns a version string indicating the version of this plugin.
+    virtual const STable& getInfo();
```
Comment (quinthar) [inline BedrockPlugin.h]: This is going to be an unsychronized access; is that fine?

File: BedrockPlugin.cpp (hunk 1 above)
Comment (tylerkaraszewski) [inline BedrockPlugin.cpp]: They could create their own `MyPluginStatus` command or something, and use that, but I don't see a problem using it here. Plugins are already married to libstuff as part of bedrock anyway.

**Pull request merged**